pkg/controller/controller.go

```diff
@@ -134,13 +140,70 @@ func (c *Controller) clusterExists(name string, cl cloudprovider.Clusters) (bool
 
 // CreateComputePool create a compute node pool.
 func (c *Controller) CreateComputePool(p model.ComputePool) error {
+	cl, impl := c.Cloud.Clusters()
+	if !impl {
+		return ErrNotImplemented
+	}
 	pooler, impl := c.Cloud.NodePooler()
 	if !impl {
 		return ErrNotImplemented
 	}
+
+	// TODO Check if a masterpool for given cluster exists first? Maybe we
+	// don't care if masterpool does not exist yet, as long as the cluster infra exists?
+	if exists, err := c.clusterExists(p.ClusterName, cl); !exists {
+		return err
+	}
+
+	// Check if a compute pool with the same name exists already.
+	computeExists, err := c.computePoolExists(p.ClusterName, p.Name, pooler)
+	if err != nil {
+		return err
+	}
+	if computeExists {
+		return fmt.Errorf("compute pool %q already exists in cluster %q", p.Name, p.ClusterName)
+	}
+
+	// Use defaults if values aren't specified.
+	if p.DiskSize == 0 {
```

Comment:
does pool `DiskSize` defaults to `0`? Or is this the way you check that property hasn't been specified?